src/CommunityToolkit/Xamarin.CommunityToolkit/Views/Popup/UWP/PopupRenderer.uwp.cs

```diff
@@ -252,14 +252,20 @@ SizeRequest IVisualElementRenderer.GetDesiredSize(double widthConstraint, double
 
 		UIElement? IVisualElementRenderer.GetNativeElement() => Control;
 
+		bool isOpen = true;
+
 		void OnDismissed(object? sender, PopupDismissedEventArgs e)
 		{
+			if (!isOpen)
+				return;
+
+			isOpen = false;
```

Comment:
Just a question: should we set ```isOpen = true``` somewhere? e.g. if I "reopen" the same popup?

Reply:
Good feedback, I'll update this to set `isOpen = true` when it displays it on the screen

Reply:
@AndreiMisiukevich I pushed some changes that set `isOpen = true` and I hooked into the UWP `Opened` event